src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java

```diff
@@ -69,4 +100,233 @@ public boolean isExcluded(ProgrammeMembership programmeMembership) {
+   *                   milestone.
+   * @param jobDataMap The map of job data.
+   * @param when       The date to schedule the notification to be sent.
+   * @throws SchedulerException if the job could not be scheduled.
+   */
+  private void scheduleNotification(String jobId, JobDataMap jobDataMap, Date when)
+      throws SchedulerException {
+    JobDetail job = newJob(NotificationService.class)
+        .withIdentity(jobId)
+        .usingJobData(jobDataMap)
+        .storeDurably(false)
+        .build();
+
+    Trigger trigger = newTrigger()
+        .withIdentity(TRIGGER_ID_PREFIX + jobId)
+        .startAt(when)
+        .build();
+
+    Date scheduledDate = scheduler.scheduleJob(job, trigger);
+    log.info("Notification for {} scheduled for {}", jobId, scheduledDate);
+  }
+
+  /**
+   * Remove a scheduled notification if it exists.
+   *
+   * @param jobId The job id key to remove.
+   * @throws SchedulerException if the scheduler failed in its duties (non-existent jobs do not
+   *                            trigger this exception).
+   */
+  private void removeNotification(String jobId) throws SchedulerException {
+    JobKey jobKey = new JobKey(jobId);
+    // Delete the job and unschedule its triggers.
+    // We do not simply remove the trigger, since a replacement job may have different data
+    // (e.g. programme name).
+    scheduler.deleteJob(jobKey);
+    log.info("Removed any stale notification scheduled for {}", jobId);
+  }
+
+  /**
+   * Get a future schedule for a notification from the programme start date and day offset.
+   *
+   * @param programmeStartDate The programme starting date.
+   * @param daysBeforeStart    The number of days prior to the programme start date.
+   * @return The notification scheduling date and time.
+   */
+  private Date getScheduleDate(LocalDate programmeStartDate, int daysBeforeStart) {
+    Date milestone;
+    LocalDate milestoneDate = programmeStartDate.minusDays(daysBeforeStart);
+    if (!milestoneDate.isAfter(LocalDate.now())) {
+      // 'Missed' milestones: schedule to be sent soon, but not immediately
+      // in case of human editing 'jitter'.
+      milestone = Date.from(Instant.now()
+          .plus(PAST_MILESTONE_SCHEDULE_DELAY_HOURS, ChronoUnit.HOURS));
+
+    } else {
+      // Future milestone.
+      milestone = Date.from(milestoneDate
+          .atStartOfDay()
+          .atZone(ZoneId.systemDefault())
+          .toInstant());
+    }
+    return milestone;
+  }
+
+  /**
+   * Helper function to determine whether a notification should be scheduled.
+   *
+   * @return true if it should be scheduled, false otherwise.
+   */
+  private boolean shouldScheduleNotification(LocalDate programmeStartDate,
+      NotificationType milestone, Map<NotificationType, Instant> notificationsAlreadySent) {
+
+    //do not resend any notification
+    if (notificationsAlreadySent.containsKey(milestone)) {
+      return false;
```

Comment:
A/C #3 :
Given that we have already sent reminder to PGDiT
When the start date is changed to be outside 8 weeks 
Then we don't send the reminder email when we get back to the 8 weeks